src/main/java/gov/nasa/cumulus/metadata/aggregator/processor/ImageProcessor.java

```diff
@@ -106,11 +116,23 @@ public boolean isDownloadUrlAlreadyExist(JsonArray relatedUrls, String downloadU
         downloadUrl = StringUtils.trim(downloadUrl);
         for (JsonElement relatdUrl : relatedUrls) {
             String umg_downloadUrl = StringUtils.trim(relatdUrl.getAsJsonObject().get("URL").getAsString());
-            if(StringUtils.compare(umg_downloadUrl, downloadUrl) ==0) return true;
+            if(StringUtils.compare(umg_downloadUrl, downloadUrl) == 0) return true;
         }
         return false;
     }
 
+    public void removeExistingUrls(JsonArray relatedUrls, String downloadUrl) {
+        downloadUrl = StringUtils.trim(downloadUrl);
+        Iterator<JsonElement> iterator = relatedUrls.iterator();
+        while (iterator.hasNext()) {
+            JsonElement relatedUrlElement = iterator.next();
+            String umg_downloadUrl = StringUtils.trim(relatedUrlElement.getAsJsonObject().get("URL").getAsString());
+            if (StringUtils.compare(umg_downloadUrl, downloadUrl) == 0) {
```

Comment:
Perhaps using compareIgnoreCase instead of compare?
compare ignoring case may see granuleId_sst.png and granuleId_SST.png (which as part of the URL string) as equal.